### PR TITLE
Docker.httpClient() + unit tests

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #165

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#166](https://github.com/amihaiemil/docker-java-api/issues/166) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#166](https://github.com/amihaiemil/docker-java-api/issues/166) is assigned to @llorllale[/z](https://www.0crat.com/u/llorllale), here is [why](https://www.0crat.com/footprint/G6LPQQV2P/552ac28c-088b-4298-9951-f2a0bc400cf8); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/com/amihaiemil/docker/LocalDockerTestCase.java [hunk 1]
```diff
@@ -126,4 +127,21 @@ public void returnsImages() {
...
+     */
+    @Test
+    public void returnsHttpClient() {
+        MatcherAssert.assertThat(
+            new LocalDocker(
+                new File("/var/run/docker.sock")
+            ).httpClient(),
+            Matchers.allOf(
+                Matchers.notNullValue(),
+                Matchers.instanceOf(HttpClient.class),
```
Comment (llorllale) [inline src/test/java/com/amihaiemil/docker/LocalDockerTestCase.java]: @amihaiemil this 'instanceOf' check seems unnecessary

File: src/test/java/com/amihaiemil/docker/RemoteDockerTestCase.java [hunk 2]
```diff
@@ -121,4 +122,23 @@ public void returnsImages() {
...
+    public void returnsHttpClient() {
+        final HttpClient client = Mockito.mock(HttpClient.class);
+        MatcherAssert.assertThat(
+            new RemoteDocker(
+                client,
+                URI.create("http://localhost")
+            ).httpClient(),
+            Matchers.allOf(
+                Matchers.notNullValue(),
+                Matchers.instanceOf(HttpClient.class),
```
Comment (llorllale) [inline src/test/java/com/amihaiemil/docker/RemoteDockerTestCase.java]: @amihaiemil same here: this 'instanceOf' check seems unnecessary

**Comment by llorllale** [comment]
@amihaiemil left you a couple of minor comments.

What really bugs me though is the premise behind this issue. What's wrong with `LocalDocker`?

**Comment by amihaiemil** [comment]
@llorllale In case of a local Docker, the HttpClient would need to be configured to work over a unix socker (which is not a trivial task, you really have to know how Apache HttpClient works). So, if we already did that, we might as well let the user get it and use it directly :D

**Comment by llorllale** [comment]
@amihaiemil I think you'll have to expose the baseURI as well

**Comment by amihaiemil** [comment]
@llorllale That won't be necessary since the user will build their own HttpRequest and they should already know the base URI. Besides, our HttpClient does not contain any base URI, only the connection information (e.g. that File socket).

**Comment by llorllale** [comment]
@amihaiemil alright, go ahead then (good to merge)

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Job was finished in 30 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The job [#166](https://github.com/amihaiemil/docker-java-api/issues/166) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**